licenseApp.js

```diff
@@ -15,16 +15,41 @@ function serverListen(server) {
     }
 }
 
+/* In practice, this service is accessed as http://localhost:3010/,
+ * but why not permit access as http://[::1]:3010 too?
+ */
+const PERMITTED_HOSTS = ['localhost', '127.0.0.1', '[::1]'];
+function isPermittedHost(msg) {
+    let host_header = msg.request_headers.get_one('Host');
+    var host;
+    if (host_header) {
+        host = host_header.slice(0, host_header.lastIndexOf(':'));
+    }
```

Comment:
What happens if no `Host` header is provided? (I am not sure that’s possible, but being robust against it seems wise.) It looks like the behaviour becomes undefined (`host` is not set, but then `indexOf(host)` is called), but I’m not familiar enough with JS to know for sure.

Reply:
`var host;` => `host === undefined`. `undefined` is not an element of the list. But you're right, this can be made clearer.

Reply:
I rewrote this to (ab)use Soup's URI parser to parse the hostname out of the header.